### ISSUE TITLE
parsing issue with test expression

### ISSUE THREAD

**Issue by JoshLitt** [koalaman/shellcheck#556]
Input file:
```shell
#!/bin/bash
file=$1
[[ ${file::1} = [-.\|/\\] ]] && return 1
```
Result (shellcheck 0.4.1):
```
In /tmp/foo.sh line 4:
[[ ${file::1} = [-.\|/\\] ]] && return 1
^-- SC1073: Couldn't parse this test expression.
                         ^-- SC1020: You need a space before the ]].
                         ^-- SC1072: Missing space before ]. Fix any mentioned problems and try again.
```
If I change the = to =~ the error goes away. But this is not a regex test, it's a glob expression test.
If I add a "*" after the rhs, the error goes away, but since this is testing a single character, that's not needed. It's not bad to work around this, but not clear why it can't be parsed.

**Comment by deryni** [comment]
What's `${file::1}` supposed to be doing there? Having a default value of `:1` for the test?

**Comment by JoshLitt** [comment]
No, it's pulling off the first character of $file. It's a form of
` ${parameter:offset:length}`

**Comment by deryni** [comment]
Ah, right. Thanks.

**Comment by JoshLitt** [comment]
I guess a using `${file:0:1}` would be more clear, but generates the same parsing issue.

**Comment by koalaman** [comment]
This was due to shortcuts in parsing `[..]` which didn't account for escaped characters. This has been fixed in 5283817. Thanks!
